### PR TITLE
Simple Payments: Internationalization and CS fixes.

### LINKED ISSUE #10334: Misused translation functions in Simple Payments widget
It looks like the `*_e()` functions were used in these situations to output variables instead of using `echo`, but the translation parts of the functions can't do anything with variables.

`modules/widgets/simple-payments/form.php`, line 101:

`<?php  esc_html_e( $instance['form_product_description'] ); ?>`

should be

`<?php echo esc_textarea( $instance['form_product_description'] ); ?>`

`modules/widgets/simple-payments/widget.php`, lines 9-11:

```
			<div class='jetpack-simple-payments-title'><p><?php esc_attr_e( $instance['form_product_title'] ); ?></p></div>
			<div class='jetpack-simple-payments-description'><p><?php  esc_html_e( $instance['form_product_description'] ); ?></p></div>
			<div class='jetpack-simple-payments-price'><p><?php esc_attr_e( $instance['form_product_price'] ); ?> <?php esc_attr_e( $instance['form_product_currency'] ); ?></p></div>
```

should be

```
			<div class='jetpack-simple-payments-title'><p><?php echo esc_html( $instance['form_product_title'] ); ?></p></div>
			<div class='jetpack-simple-payments-description'><p><?php echo esc_html( $instance['form_product_description'] ); ?></p></div>
			<div class='jetpack-simple-payments-price'><p><?php echo esc_html( $instance['form_product_price'] . ' ' . $instance['form_product_currency'] ); ?></p></div>
```

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #10334

#### Changes proposed in this Pull Request:

- Do not use translation functions to return vars.
- add missing textdomain
- Various coding standards fixes.

#### Testing instructions:

1. Start wit a site using a Premium plan.
2. Go to Appearance > Customize > Widgets, and attempt to set up and make changes to a Simple Payments widget.

#### Proposed changelog entry for your changes:
* Simple Payments: Internationalization fixes.

### CODE REVIEW

File: modules/widgets/simple-payments/form.php [hunk 1]
```diff
@@ -10,23 +18,23 @@ class="widefat jetpack-simple-payments-widget-title"
 		value="<?php echo esc_attr( $instance['title'] ); ?>" />
 </p>
 <p class="jetpack-simple-payments-products-fieldset" <?php if ( empty( $product_posts ) ) { echo 'style="display:none;"'; } ?>>
-	<label for="<?php echo $this->get_field_id('product_post_id'); ?>">
+	<label for="<?php echo intval( $this->get_field_id( 'product_post_id' ) ); ?>">
```
Comment (zinigor) [inline modules/widgets/simple-payments/form.php]: What do you think about using `esc_attr` instead? And integer as a post ID is quite a safe assumption, but it is an assumption anyway.

File: modules/widgets/simple-payments/form.php [hunk 2]
```diff
@@ -10,23 +18,23 @@ class="widefat jetpack-simple-payments-widget-title"
...
 		<?php esc_html_e( 'Select a Simple Payments Button:', 'jetpack' ); ?>
 	</label>
 	<select
 		class="widefat jetpack-simple-payments-products"
-		id="<?php echo $this->get_field_id('product_post_id'); ?>"
-		name="<?php echo $this->get_field_name('product_post_id'); ?>">
+		id="<?php echo intval( $this->get_field_id( 'product_post_id' ) ); ?>"
+		name="<?php echo intval( $this->get_field_name( 'product_post_id' ) ); ?>">
```
Comment (zinigor) [inline modules/widgets/simple-payments/form.php]: Same comment as above ^

File: modules/widgets/simple-payments/form.php [hunk 3]
```diff
@@ -43,14 +51,14 @@ class="widefat jetpack-simple-payments-products"
...
+		id="<?php echo esc_attr( $this->get_field_id( 'form_action' ) ); ?>"
+		name="<?php echo esc_attr( $this->get_field_name( 'form_action' ) ); ?>"
 		value="<?php echo esc_attr( $instance['form_action'] ); ?>"
 		class="jetpack-simple-payments-form-action" />
 	<input
 		type="hidden"
-		id="<?php echo $this->get_field_id('form_product_id'); ?>"
-		name="<?php echo $this->get_field_name('form_product_id'); ?>"
+		id="<?php echo absint( $this->get_field_id( 'form_product_id' ) ); ?>"
+		name="<?php echo absint( $this->get_field_name( 'form_product_id' ) ); ?>"
```
Comment (zinigor) [inline modules/widgets/simple-payments/form.php]: Same here, what do you think about `esc_attr`?

**Review by zinigor — CHANGES_REQUESTED** [review]
Overall looks good, but I have a couple of comments. Can you please take a look?

File: modules/widgets/simple-payments/form.php (hunk 1 above)
Comment (jeherve) [reply modules/widgets/simple-payments/form.php]: Works for me. Fixed in a09ccd2614ef9ac8e413d65aeb900b584f505d7d

**Review by zinigor — APPROVED** [review]

**Comment by kraftbj** [comment]
Deployed to wp.com via r182054-wpcom.

**Pull request merged**